src/Courses/CourseDetailsTabs/CourseUsersList.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { ListGroup, ListGroupItem, Row, Col, Form, FormControl, Button } from 'react-bootstrap';
+import { UserObject } from '../CourseInterfaces';
+import _ from 'lodash';
+
+interface CourseUsersListProps {
+    users: Array<UserObject>;
+    activeUsers: Array<Set<number>>;
+    setActive: React.Dispatch<React.SetStateAction<Set<number>[]>>;
+}
+
+export const CourseUsersList: React.FC<CourseUsersListProps> = ({users, activeUsers, setActive}) => {
+    // TODO: This should be refactored into a reusable TwoState button.
+    const [selectAllState, setSelectAllState] = useState(false);
+
+    // Add or remove a user's ID to the active users array.
+    const onClickStudent = (id: number) => {
+        if (activeUsers[0].has(id)) {
+            activeUsers[0].delete(id);
+            setActive([activeUsers[0]]);
+        } else {
+            setActive([activeUsers[0].add(id)]);
+        }
+    };
+
+    const onSelectAll = () => {
+        // If all are currently selected, select none.
+        if (selectAllState) {
+            setActive([new Set()]);
+        } else {
+            const allUserIds: Array<number> = users.map(user => user.id);
+            setActive([new Set(allUserIds)]);
```

Comment:
to avoid double array should we do `...allUserIds`

Reply:
The `Set` constructor accepts an array.